### DISCUSSION TITLE
Open Github repository in browser

### DISCUSSION THREAD

**Discussion by Alarid** [johnlindquist/kit#203] [Share]
I created a simple script to open one of my Github repositories in the browser. It starts by fetching the repositories, and ask which one to open. Obviously, you would need to change `Alarid` to your GitHub username if you would like to use this script 😊

~~Also, if someone knows how to make a "loading state" to avoid having just an empty prompt while fetching, that would be awesome! It only takes a few seconds, but still...~~

UPDATE: thanks @johnlindquist for the tip on the loading state, here is the updated version:

```javascript
// Menu: Open Github Repo
// Description: Open the repo page in the browser

const USERNAME = "Alarid"

let dots = 0
let placeholderIntervalId = setInterval(() => {
  setPlaceholder(`Loading ${USERNAME} github repos`.padEnd(++dots, "."))
}, 100)

const response = await get(`https://api.github.com/users/${USERNAME}/repos`)
const repositories = response.data
clearInterval(placeholderIntervalId)

if (!repositories) exit(1)

const choice = await arg(
  "Which project?",
  repositories.map((repo) => ({
    name: repo.name,
    description: repo.description,
    value: repo.html_url,
  }))
)

exec(`open ${choice}`)

```

**Comment by johnlindquist** [comment]
You can use `setPlaceholder` and `setHint` in whatever way you want. Here's an incrementing dots solution:

```js
let dots = 0
let clearId = setInterval(() => {
  setPlaceholder("".padEnd(++dots, "."))
}, 100)

const response = await get(
  "https://api.github.com/users/Alarid/repos"
)

clearInterval(clearId)
```

I'm considering APIs for loaders, but don't have anything to show/talk about yet.

**Comment by theianjones** [comment]
Nice,

I added this so that you could type the name if you wanted or hit enter for the default:

```js
const USERNAME = await arg('Enter the github username: ', async (input) => {
  return input || 'theianjones'
})
```

**Comment by letsandeepio** [comment]
This is amazing! is there private repo version :D

**Reply by Alarid** [reply]
It's a lot more complicated, you can [check out this discussion to learn more](https://github.com/johnlindquist/kit/discussions/252)
